### ISSUE TITLE
How to use custom datasets to test the performance of a model? 

### ISSUE THREAD

**Issue by weekiiy** [Tsinghua-MARS-Lab/DenseTNT#58]
How to use custom datasets to test the performance of a model? I did not see the test module in the code.I want to do trajectory prediction on my own dataset to test the performance of the model, but I couldn't find the testing module in the code. How to use your own dataset for testing and prediction?

**Comment by GentleSmile** [comment]
The testing module loads the dataset [here](https://github.com/Tsinghua-MARS-Lab/DenseTNT/blob/294596b0b141fb7b18248889e5c409630c5f7ed6/src/do_eval.py#L60). You need to modify the [dataset](https://github.com/Tsinghua-MARS-Lab/DenseTNT/blob/294596b0b141fb7b18248889e5c409630c5f7ed6/src/dataset_argoverse.py#L695) to your own format.

**Comment by weekiiy** [comment]
Thank you for your reply. If I want to modify the map information that matches the training set and the validation set, how to do so and how to modify it.

**Comment by GentleSmile** [comment]
Map information is converted into vectors during training and inference. You need to modify [map processing](https://github.com/Tsinghua-MARS-Lab/DenseTNT/blob/294596b0b141fb7b18248889e5c409630c5f7ed6/src/dataset_argoverse.py#L43) to support your own map format.

**Comment by weekiiy** [comment]
Thank you for your reply, which effectively solved my above problems. I have the following questions
1. If I want to apply this code for trajectory prediction in a certain area, there are already vehicle trajectory data sets in this area, but the editing of the map becomes difficult. If it's a 20m square of empty space, how do we build this simple map and apply it to our algorithm?
2. What if I only want to use my own map data during testing?
3. What if I want to change the density of the target point?
This model is really excellent, but I did encounter the above problems in application, thank you for your patience to answer

**Comment by 1Nsan5** [comment]
> Thank you for your reply, which effectively solved my above problems. I have the following questions
> 
> 1. If I want to apply this code for trajectory prediction in a certain area, there are already vehicle trajectory data sets in this area, but the editing of the map becomes difficult. If it's a 20m square of empty space, how do we build this simple map and apply it to our algorithm?
> 2. What if I only want to use your own map data during testing?
> 3. What if I want to change the density of the target point?
>    This model is really excellent, but I did encounter the above problems in application, thank you for your patience to answer
